src/Tests/Microsoft.NET.Publish.Tests/GivenThatAPublishedDepsJsonShouldContainVersionInformation.cs

```diff
@@ -0,0 +1,273 @@
+            }
+
+            var immutableRuntimeLibrary = dependencyContext.RuntimeLibraries.Single(l => l.Name == "System.Collections.Immutable");
+            var immutableRuntimeFiles = immutableRuntimeLibrary.RuntimeAssemblyGroups.SelectMany(rag => rag.RuntimeFiles).ToList();
+            immutableRuntimeFiles.Should().NotBeEmpty();
+            foreach (var runtimeFile in immutableRuntimeFiles)
+            {
+                runtimeFile.AssemblyVersion.Should().Be("1.2.3.0");
+                runtimeFile.FileVersion.Should().Be("4.6.26216.2");
+            }
+        }
+
+        [Fact]
+        public void Versions_are_not_included_for_self_contained_apps()
+        {
+            Versions_are_not_included(false);
+        }
+
+        [Fact]
+        public void Versions_are_not_included_for_build()
+        {
+            Versions_are_not_included(true);
+        }
+
+        private void Versions_are_not_included(bool build, [CallerMemberName] string callingMethod = "")
+        {
+            var testProject = GetTestProject();
+            testProject.RuntimeIdentifier = EnvironmentInfo.GetCompatibleRid(testProject.TargetFrameworks);
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, callingMethod)
+                .Restore(Log, testProject.Name);
+
+            NuGetConfigWriter.Write(testAsset.TestRoot, NuGetConfigWriter.DotnetCoreMyGetFeed);
+
+            MSBuildCommand command;
+            if (build)
+            {
+                command = new BuildCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            }
+            else
+            {
+                command = new PublishCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            }
+
+            command.Execute()
+                .Should()
+                .Pass();
+
+            var outputDirectory = command.GetOutputDirectory(testProject.TargetFrameworks, runtimeIdentifier: testProject.RuntimeIdentifier);
+            outputDirectory.Should().HaveFile(testProject.Name + ".deps.json");
+
+            DependencyContext dependencyContext;
+            using (var depsJsonFileStream = File.OpenRead(Path.Combine(outputDirectory.FullName, $"{testProject.Name}.deps.json")))
+            {
+                dependencyContext = new DependencyContextJsonReader().Read(depsJsonFileStream);
+            }
+
+            var allRuntimeFiles = dependencyContext.RuntimeLibraries.SelectMany(rl => rl.NativeLibraryGroups.Concat(rl.RuntimeAssemblyGroups))
+                .SelectMany(rag => rag.RuntimeFiles);
+
+            allRuntimeFiles.Should().NotBeEmpty();
+
+            foreach (var runtimeFile in allRuntimeFiles)
+            {
+                runtimeFile.AssemblyVersion.Should().BeNull();
+                runtimeFile.FileVersion.Should().BeNull();
+            }
+        }
+
+        [Fact(Skip = "Host deps.json doesn't have runtime file version info yet: https://github.com/dotnet/sdk/issues/2124")]
```

Comment:
Did you try mocking the metadata (to see if it will pass when the metadata is available)?